File: app/models/item.rb
class Item < ActiveRecord::Base
  attr_accessible :description, :condition, :guaranty, :guaranty_service, :status, :location_id, :price
  
  has_one :model_order
  #has_one :order, through: :model_order
  #has_one :model, through: :model_order
  belongs_to :order
  belongs_to :model
  belongs_to :distributor
  belongs_to :vendor
  belongs_to :product_type
  belongs_to :product_invoice
  belongs_to :location, :class_name => "Equipment"
  
  #validates :price, :format => { :with => /^\d+??(?:\.\d{0,2})?$/ }, :numericality =>{:greater_than => 0}
  
end

File: public/app/model/inventory/Item.js
Ext.define('coloMS.model.inventory.Item', {
    extend: 'coloMS.model.inventory.Base',
    
    fields: [
        {
            name: 'vendor_name',
            type: 'string'
        },    
        {
            name: 'product_type',
            type: 'string'
        },
        {
            name: 'model_name',
            type: 'string',
        },
        {
            name: 'model_power',
            type: 'int',
        },        
        {
            name: 'order_id',
            type: 'string',
        },
        {
            name: 'condition',
            type: 'string'
        },
        {
            name: 'guaranty',
            type: 'string'
        },
        {
            name: 'guaranty_service',
            type: 'string'
        },                
        {
            name: 'status',
            type: 'string'
        },                
        {
            name: 'location_id',
            type: 'int'
        },  
        {
            name: 'price',
            type: 'float'
        },
        {
            name: 'old_id',
            type: 'int'
        },                                   

        // decorate releashions
         {
            name: '_location',
            type: 'string'
        }           
    ]
});